### ISSUE TITLE
Streamable templates

### ISSUE THREAD

**Issue by evroon** [Keats/tera#953]
I found this issue: https://github.com/Keats/tera/issues/211 but it has been closed for the last 5 years without update.

So I wondered whether this is something that's still being considered. I think it makes a lot of sense to support streaming, both for better user experience in for example webserver responses, but most of all as it reduces RAM usage.

minijinja does [support streaming](https://github.com/mitsuhiko/minijinja/blob/main/examples/streaming/src/main.rs) for example, but I would prefer not to use minijinja for other reasons.

**Comment by Keats** [comment]
Does https://docs.rs/tera/latest/tera/struct.Tera.html#method.render_to not work for you?

**Comment by evroon** [comment]
Ah that solves the second part of my problem, but is there a way to pass variables of type iterator to the template's context?

**Comment by Keats** [comment]
No, not currently. It could maybe be added to v2 (https://github.com/Keats/tera2) but that would a significant amount of work that I wouldn't have time to do myself

**Comment by evroon** [comment]
Ah okay, understandable, thanks for letting me know

**Comment by Keats** [comment]
Closing as part of the Tera v2 release. If this is still relevant in v2, please reopen — see the 
  migration guide: https://github.com/Keats/tera/blob/master/MIGRATION.md
